compiler/rustc_serialize/src/raw.rs

```diff
@@ -0,0 +1,678 @@
+use crate::serialize::{self, Decoder as _, Encoder as _};
+use std::borrow::Cow;
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::{self, Write};
+use std::mem::MaybeUninit;
+use std::path::Path;
+use std::ptr;
+
+// -----------------------------------------------------------------------------
+// Encoder
+// -----------------------------------------------------------------------------
+
+pub type EncodeResult = Result<(), !>;
+
+pub struct Encoder {
+    pub data: Vec<u8>,
+}
+
+impl Encoder {
+    pub fn new(data: Vec<u8>) -> Encoder {
+        Encoder { data }
+    }
+
+    pub fn into_inner(self) -> Vec<u8> {
+        self.data
+    }
+
+    #[inline]
+    pub fn position(&self) -> usize {
+        self.data.len()
+    }
+}
+
+macro_rules! write_raw {
+    ($enc:expr, $value:expr, $int_ty:ty) => {{
+        let bytes = $value.to_ne_bytes();
```

Comment:
Can you please use `to_le_bytes` instead? This is just as fast on little endian systems, but for example makes it possible to move the incr cache to a big endian system and use then `--target`.

Reply:
Is this an actual use case? This PR makes the file format unportable (because of `isize`/`usize` mainly), with the objective to memmap part of it. If portability is required, I need to change the implementation to make sure of it.

Reply:
Right I can't name any. I do remember some talk about maybe changing crate metadata to be an incr comp snapshot or something like that in the future. I can't remember the details. In that case portability is very important for cross-compilation.